File: ios/Ember/Core/Models/MemoryModels.swift
import Foundation

/// A single memory item from the Mem0 memory layer.
/// Maps to the backend's `MemoryItem` Pydantic schema.
/// Decoded with `JSONDecoder.ember` (snake_case -> camelCase).
struct MemoryItem: Codable, Identifiable, Equatable {
    let id: String
    let memory: String
    let createdAt: Date?
}

/// Response from `GET /api/v1/characters/:id/memories` and `GET /api/v1/memories`.
/// Maps to the backend's `MemoryListResponse` Pydantic schema.
struct MemoryListResponse: Codable {
    let memories: [MemoryItem]
}

File: android/app/src/main/java/ai/ember/app/core/models/MemoryModels.kt
package ai.ember.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * A single memory entry from Mem0, mapped from GET /api/v1/memories
 * and GET /api/v1/characters/:id/memories responses.
 */
@Serializable
data class MemoryItem(
    val id: String,
    val memory: String,
    @SerialName("created_at") val createdAt: String? = null,
)

/**
 * Response wrapper for memory list endpoints.
 */
@Serializable
data class MemoryListResponse(
    val memories: List<MemoryItem>,
)
